interface/patient_file/encounter/Services/ViewHelper.php

```diff
@@ -0,0 +1,264 @@
+<?php
+/**
+ * Helper functions for an encounter view files
+ *
+ * LICENSE: This program is free software; you can redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation; either version 3 of the License, or (at your option)
+ * any later version. This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public
+ * License for more details. You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <http://opensource.org/licenses/gpl-license.php>;.
+ *
+ * @package OpenEMR
+ * @subpackage Encounter
+ * @license GNU GPL3
+ * @author Robert Down <robertdown@live.com>
+ * @copyright Copyright (C) 2017 Robert Down
+ */
+
+namespace OpenEMR\Encounter\Services;
+
+class ViewHelper
+{
+
+    private $oldCategory = '';
+
+    static function getRegistry($state = "1", $limit = "unlimited", $offset = "0")
+    {
+        global $attendant_type;
+        $sql = "SELECT * FROM registry WHERE {column}= 1 AND state LIKE ? ORDER BY category, priority, name";
+        $params = array();
+        $column = ($attendant_type == 'pid') ? 'patient_encounter' : 'therapy_group_encounter';
+        $sql = str_replace("{column}", $column, $sql);
+        $params[1] = $state;
+        if ($limit != "unlimited") {
+            $params[2] = $limit;
+            $params[3] = $offset;
+            $sql .= " LIMIT ?, ?";
+        }
+        $result = sqlStatement($sql, $params);
+        if ($result) {
+            $all = array();
+            while ($row = sqlFetchArray($result)) {
+                array_push($all, $row);
+            }
+            return $all;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Create a bootstrap-based navbar based on array
+     *
+     * $elements = array(
+     *     'name' => 'Text displayed to user in link. Required',
+     *     'href' => 'Location of menu item. Required',
+     *     'atts' => ['Array of valid HTML5 attributes to attach to the anchor tag'],
+     *     'linkClass' => 'Space separated string of extra classes to add to <a> element. Optional',
+     *     'listItemClass' => 'Space separated string of extra classes to add to <li> element. Optional',
+     *     'subItems' => array('Recursion of $elements array structure. Optional'));
+     *
+     * @todo This does not actually work, the links are not yet hooked up. RD 2017-04-19
+     * @todo Handling the subItems list could be better - needs to be truly recursive. RD 2017-04-23
```

Comment:
Whoops, line 62 is complete, I'll pull it. Line 63 is still in effect. Code is functional, but could be better